spec/components/filter_summary_spec.rb

```diff
@@ -0,0 +1,85 @@
+require "spec_helper"
+
+describe "Filter summary component", type: :view do
+  def component_name
+    "filter_summary"
+  end
+
+  let(:filters) do
+    [
+      {
+        label: "Filter 1",
+        value: "Value 1",
+        remove_href: "/remove_url",
+        visually_hidden_prefix: "Remove filter",
+      },
+      {
+        label: "Filter 2",
+        value: "Value 2",
+        remove_href: "/remove_url",
+        visually_hidden_prefix: "Remove filter",
+      },
+      {
+        label: "Filter 3",
+        value: "Value that is so long that the styling needs to handle it correctly",
+        remove_href: "/remove_url",
+        visually_hidden_prefix: "Remove filter",
+      },
+    ]
+  end
+
+  def render_component(locals)
+    render "components/#{component_name}", locals
+  end
+
+  it "does not render a filter summary when no filters array passed" do
+    render_component({ heading_text: "Selected filters" })
+
+    assert_select ".app-c-filter-summary__heading", false
```

Comment:
Given we render nothing, we might want to check for the absence of the whole `.app-c-filter-summary`